interface/src/lib/stores/socket.ts

```
@@ -47,16 +47,35 @@ function createWebSocket() {
-			if (event) listeners.get(event)?.forEach((listener) => listener(data));
-			else listeners.get("monitor")?.forEach((listener) => listener(new Uint8Array(message.data))); // 🌙 if no event, assume monitor data (see emitEvent char * output)
+			
+			// Safe destructuring
+			const { event = null, data = null } = payload;
+			
+			if (event) {
+				if (data !== null && data !== undefined) {
+					listeners.get(event)?.forEach((listener) => listener(data));
+				}
+			} else { // 🌙 if no event, assume monitor data (see emitEvent char * output)
+				listeners.get("monitor")?.forEach((listener) => 
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)